### PR TITLE
Ref #276: Propose an integration test framework

### LINKED ISSUE #276: Propose an integration test framework
Integration tests need to be added to at least cover nominal use cases to validate that the camel features work properly. But to avoid duplicating a lot of code, a small integration test framework should be proposed that must work with and without test containers.

### PR DESCRIPTION
**Pull request by essobedo**
fixes #276 

* Provide a way to easily deploy Camel routes written in Java by implementing `CamelRouteSupplier`
* Provide a way to easily deploy Camel routes written using Blueprint DSL by adding them into `OSGI-INF/blueprint`
* Allow remote debug by adding the system property `camel.karaf.itest.debug` to the build command
* Allow providing hints to the framework thanks to the annotation `CamelKarafTestHint`
* Allow to manage several Camel contexts
* Provide samples to show how to use it

### CODE REVIEW

File: features/src/main/feature/camel-features.xml [hunk 1]
```diff
@@ -828,12 +828,18 @@
...
+        <bundle dependency="true">mvn:org.glassfish.hk2/osgi-resource-locator/2.5.0-b42</bundle>
+        <bundle dependency="true">mvn:org.eclipse.parsson/parsson/${parson-version}</bundle>
+        <bundle dependency='true'>mvn:jakarta.json/jakarta.json-api/${jakarta-json-api-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.opentelemetry/opentelemetry-api/${opentelemetry-version}</bundle>
+        <bundle dependency='true'>wrap:mvn:io.opentelemetry/opentelemetry-context/${opentelemetry-version}</bundle>
         <bundle dependency='true'>wrap:mvn:org.apache.httpcomponents/httpasyncclient/${httpasyncclient-version}</bundle>
-        <bundle dependency='true'>wrap:mvn:co.elastic.clients/elasticsearch-java/8.12.1</bundle>
-        <bundle dependency='true'>wrap:mvn:org.elasticsearch.client/elasticsearch-rest-client/8.12.1</bundle>
-        <bundle dependency='true'>wrap:mvn:org.elasticsearch.client/elasticsearch-rest-client-sniffer/8.12.1</bundle>
+        <bundle dependency='true'>wrap:mvn:co.elastic.clients/elasticsearch-java/${elasticsearch-java-client-version}</bundle>
```
Comment (jbonofre) [inline features/src/main/feature/camel-features.xml:840]: I would prefer to not mix unrelated changes: this is not related directly to the integration test.

File: tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/ExternalResource.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package org.apache.karaf.camel.itests;
```
Comment (jbonofre) [inline tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/ExternalResource.java:1]: ASF header is missing there.

File: features/src/main/feature/camel-features.xml (hunk 1 above)
Comment (essobedo) [reply features/src/main/feature/camel-features.xml:840]: I agree, but we need real tests to validate the "framework" and while testing the camel ES feature, we realized that it doesn't work as it is. But no worries, if really needed, I will split it in 2 later today

Comment (jbonofre) [reply features/src/main/feature/camel-features.xml:840]: That's OK, just a note for next PRs ;)

File: tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/ExternalResource.java (hunk 2 above)
Comment (essobedo) [reply tests/camel-integration-test/src/main/java/org/apache/karaf/camel/itests/ExternalResource.java:17]: Good catch thx 👍

File: features/src/main/feature/camel-features.xml (hunk 1 above)
Comment (essobedo) [reply features/src/main/feature/camel-features.xml:840]: ok 👍

**Review by f2par0 — APPROVED** [review]
LGTM!

**Review by jbonofre — APPROVED** [review]
